internal/pkg/service/common/etcdop/watch_test.go

```diff
@@ -222,6 +222,130 @@ func TestPrefix_GetAllAndWatch(t *testing.T) {
 	assert.False(t, ok)
 }
 
+func TestPrefix_GetAllAndWatch_ErrCompacted(t *testing.T) {
+	t.Parallel()
+
```

Comment:
A tu je novy unit test, ktory generuje `ErrCompacted` error pocas `GetAllAndWatch`.
Je tam custom `dialer`, z ktoreho odchytime spojenie a pocas testu ho ukoncime.
`watcher` sa snazi vytvorit nove spojenie, ale to zablokujeme cez `dialerLock`.

Pocas vypadku pridame dalsie kluce a spravit `Compact` operaciu.
Ked sa `watcher` znova pripoji, tak dostane `ErrCompacted`, a testujeme, ze sa spravne vygeneruje `Restarted` event, a ziskame opat vsetky kluce od zaciatku.